### PR TITLE
Add Sombrero

### PR DESCRIPTION
**Pull request by githoneybadger**
Adds a fancy orange sombrero to the list of available hats.

### CODE REVIEW

File: data/json/items/armor.json [hunk 1]
```diff
@@ -5983,6 +5983,29 @@
...
+        "price" : 5500,
+        "material" : ["cotton", "null"],
+        "volume" : 14,
+        "cutting" : 0,
+        "warmth" : 1,
+        "phase" : "solid",
+        "environmental_protection" : 4,
+        "encumbrance" : 20,
+        "//" : "Head encumbrance stops you from wearing additional headgear.  No helmet over a sombrero, sorry.  But brimmed gear gets the anti-glare flag!",
+        "flags" : ["FANCY", "BELTED", "SUN_GLASSES"],
```
Comment (Coolthulhu) [inline data/json/items/armor.json]: Should be OUTER, not BELTED.

**Comment by chaosvolt** [comment]
Orange now exists as a color? Hell yes. owo

**Comment by BevapDin** [comment]
> Orange now exists as a color? Hell yes. owo

Actually, no. That shouldn't work. The game should ignore the color definition and there should be a `debugmsg` or.

**Comment by chaosvolt** [comment]
Dangit. I've been meaning to fix that. Does it need anything more than a definition in Data/Raw/Colors.json?

File: data/json/items/armor.json (hunk 1 above)
Comment (githoneybadger) [inline data/json/items/armor.json]: K, I'll set that.  It seems a bit odd in game to see "this gear is generally worn over clothing" with the OUTER flag but no big either way.

**Comment by githoneybadger** [comment]
Yeah I looked at it too.  It appears the bit count is 8, so the palette could be 256 color total....instead of the 16 it's currently set to.  I may expand this in the future but am new to codebase so perhaps not feasible, this was just an initial review.

Sorry, I had never noticed that items had colors before, it was showing as grey before as a default.  For right now, it's set to red, which is an ingame color.

**Comment by chaosvolt** [comment]
It seems light red is the norm for orange things that aren't orange. Yellow would work too, since that's the norm for straw items.

I was juuust starting to try a PR for it, when I realized that the color files in source have NUMBERED settings for damn near every combo of existing color. No way in hell am I going to get far trying to mess with that. ;w;

**Comment by githoneybadger** [comment]
Light red it is.  TY!

**Comment by Rivet-the-Zombie** [comment]
Since it's made of straw, shouldn't its material be ````veggy````?

**Comment by githoneybadger** [comment]
Clarified it's a felt sombrero, not straw, in desc.

**Comment by chaosvolt** [comment]
Hmm. Not sure if that'd mean wool or fur then. o3o

**Comment by githoneybadger** [comment]
Wool, usually.  Ok, changed.

**Pull request merged**